### PR TITLE
2464 - Integrates Chrome Components with TinaAdmin

### LINKED ISSUE #2464: Integrating Chrome Components into TinaAdmin
**Depends on the Chrome Components PR: https://github.com/tinacms/tinacms/pull/2434**

When `TinaAdmin` was initially written, we wanted to use Tailwind, so much of `TinaAdmin`'s look and feel is custom and built specifically for Tailwind.

Now that Tailwind is integrated into Tina, we can pull components from `@tinacms/toolkit` and use them properly - specifically, the updated `Nav`, `FormBuilder`, and overall layout of the fullscreen page.

### PR DESCRIPTION
**Pull request by Enigmatical**
As part of our push to unify the Sidebar experience with TinaAdmin, TinaAdmin will now make use of _some_ of the Sidebar's elements (mostly just the `Nav`).

Additional work has been done to match the UI feel from the demo website with actual TinaAdmin.

Closes #2464

### CODE REVIEW

**Comment by Enigmatical** [comment]
#### Remaining Punchlist:
- [ ] Address the multiple form bug that Logan and I found: https://github.com/tinacms/tinacms/issues/2505
- [ ] Add a triple-dot sub-menu to each list item (for "View", "Edit in Admin" (later), and "Delete" (later)
- [ ] Provide support for `RouteMapper`
  - For this phase, we'll just tuck the "View in Website" link inside a triple-dot sub-menu on each list item
  - Next phase, we'll actually link to the pages via the main link
- [ ] Fix the `ScreenPage` for `_type: screen` plugins based on `layout`

File: packages/@tinacms/toolkit/src/index.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ export type { ContentCreatorPlugin } from './packages/forms'
 export * from './packages/forms'
 export * from './packages/icons'
 export * from './packages/react-dismissible'
+export { Nav } from './packages/react-sidebar'
```
Comment (Enigmatical) [inline packages/@tinacms/toolkit/src/index.ts:42]: This is the only component that needs to be exported to `TinaAdmin`.  The rest of the `Sidebar` (and its context) are not necessary.

File: packages/@tinacms/toolkit/src/packages/react-sidebar/components/Nav.tsx [hunk 2]
```diff
@@ -19,29 +19,38 @@ limitations under the License.
...
   userName?: string
+  collections?: { label: string; name: string }[]
   contentCreators?: any
-  screens?: any
+  screens?: ScreenPlugin[]
+  sidebarWidth?: number
+  RenderNavSite: React.ComponentType<{ view: ScreenPlugin }>
+  RenderNavCollection: React.ComponentType<{
+    collection: { label: string; name: string }
+  }>
```
Comment (Enigmatical) [inline packages/@tinacms/toolkit/src/packages/react-sidebar/components/Nav.tsx:40]: The majority of the changes to `Nav` are based around these new props.  I effectively decoupled `Nav` from `SidebarContext`, freeing me from having to maintain `SidebarContext` in the other place I wanted to use `Nav`.

Instead, I pass in the behaviors that I want `Nav` to use.

File: packages/tinacms/src/auth/TinaCloudProvider.tsx [hunk 3]
```diff
@@ -173,6 +174,14 @@ export const TinaCloudProvider = (
...
     }
   }
+  /**
+   * Attaches a new instance of the TinaAdminApi to `cms.api.admin`
+   */
+  const setupAdmin = async () => {
+    if (!cms.api.admin) {
+      cms.registerApi('admin', new TinaAdminApi(cms))
+    }
+  }
```
Comment (Enigmatical) [inline packages/tinacms/src/auth/TinaCloudProvider.tsx:184]: Register the `admin` API alongside the `tina` one for making calls to `TinaAdminApi` (specifically for `fetchCollections()` in the `Nav`).

File: packages/tinacms/src/auth/TinaCloudProvider.tsx [hunk 4]
```diff
@@ -189,6 +198,9 @@ export const TinaCloudProvider = (
   }
 
   setupMedia()
+  if (cms.flags.get('tina-admin') === true) {
+    setupAdmin()
+  }
```
Comment (jeffsee55) [inline packages/tinacms/src/auth/TinaCloudProvider.tsx:203]: I realize we've been doing this with `setupMedia,` but should this be inside `useMemo`?

File: packages/tinacms/src/admin/pages/CollectionListPage.tsx [hunk 5]
```diff
@@ -203,4 +223,59 @@ const CollectionListPage = () => {
...
+              enterTo="transform opacity-100 translate-y-0"
+              leave="transition duration-75 ease-in"
+              leaveFrom="transform opacity-100 translate-y-0"
+              leaveTo="transform opacity-0 -translate-y-2"
+            >
+              <Menu.Items className="bg-white border border-gray-150 rounded-lg shadow-lg">
+                {items.map((item) => {
+                  const Icon = item.icon ? item.icon : BiExit
+                  return (
+                    <Menu.Item>
```
Comment (jeffsee55) [inline packages/tinacms/src/admin/pages/CollectionListPage.tsx:257]: Looks like we need a `key` here, guessing `item.label` would suffice

**Review by jeffsee55 — CHANGES_REQUESTED** [review]
This looks awesome! Feels really good. I'll put some wishlist-type of items over on the next phase's issue https://github.com/tinacms/tinacms/issues/2530 so just a couple of small comments here

File: packages/tinacms/src/auth/TinaCloudProvider.tsx [hunk 6]
```diff
@@ -189,6 +198,9 @@ export const TinaCloudProvider = (
   }
 
   setupMedia()
+  if (cms.flags.get('tina-admin') === true) {
+    setupAdmin()
```
Comment (jeffsee55) [inline packages/tinacms/src/auth/TinaCloudProvider.tsx:202]: I know we're doing the same thing with `setupMedia`, but should this be inside a `useMemo` hook?

**Review by jeffsee55 — DISMISSED** [review]

**Review by jeffsee55 — APPROVED** [review]

**Pull request merged**